### PR TITLE
CFE-3535: Added unit test GitHub Action

### PR DESCRIPTION
**Pull request by olehermanse**

### CODE REVIEW

File: .github/workflows/build_and_unit_test.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (vpodzime) [inline .github/workflows/build_and_unit_test.yml:11]: Booooo :-1:

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

File: .github/workflows/build_and_unit_test.yml (hunk 1 above)
Comment (olehermanse) [reply .github/workflows/build_and_unit_test.yml:11]: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on

@vpodzime Which one do you want?

Comment (vpodzime) [reply .github/workflows/build_and_unit_test.yml:11]: :disappointed: 

```
Ubuntu 20.04 | ubuntu-20.04
Ubuntu 18.04 | ubuntu-latest or ubuntu-18.04
```
??? Let's at least use the latest Ubuntu and not *ubuntu-latest*.

Comment (olehermanse) [reply .github/workflows/build_and_unit_test.yml]: Changed to Ubuntu 20.04

**Review by vpodzime — APPROVED** [review]
ACK

**Review by larsewi — APPROVED** [review]
Awesome !

**Pull request merged**